src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/domain/DecisionAndActionsTranslation.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.domain
+
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.entity.DecisionAndActions
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.entity.ReferenceData
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.enumeration.Reason
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.enumeration.Source
+import uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.model.request.CreateDecisionAndActionsRequest
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+fun DecisionAndActions.toModel() =
+  uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.model.DecisionAndActions(
+    decisionConclusion,
+    decisionOutcome.toReferenceDataModel(),
+    decisionOutcomeSignedOffBy?.toReferenceDataModel(),
+    decisionOutcomeRecordedBy,
+    decisionOutcomeRecordedByDisplayName,
+    decisionOutcomeDate,
+    nextSteps,
+    actionOpenCsipAlert,
+    actionNonAssociationsUpdated,
+    actionObservationBook,
+    actionUnitOrCellMove,
+    actionCsraOrRsraReview,
+    actionServiceReferral,
+    actionSimReferral,
+    actionOther,
+  )
+
+fun CreateDecisionAndActionsRequest.toCsipRecordEntity(
+  referral: Referral,
```

Comment:
There's duplication here between the extension method request property and the parameters being passed in. You don't need the latter. Most of these properties can be accessed directly from the request as with `conclusion` rather than the passed in `decisionConclusion`.

It's also the case that this extension method doesn't really add anything. You could call createDecisionAndActions directly from the service.